src/main/java/ca/bc/gov/backendstartapi/service/OrchardService.java

```diff
@@ -44,17 +66,129 @@ public Optional<OrchardLotTypeDescriptionDto> findNotRetiredOrchardValidLotType(
+    }
+
+    // Orchard
+    OrchardParentTreeDto orchardParentTreeDto = new OrchardParentTreeDto();
+    orchardParentTreeDto.setOrchardId(orchard.get().getId());
+    orchardParentTreeDto.setVegetationCode(orchard.get().getVegetationCode());
+    orchardParentTreeDto.setSeedPlanUnitId(spuId);
+
+    long endingTwo = Instant.now().toEpochMilli();
+    log.info("Time elapsed creating basic OrchardParentTreeDto: {}", endingTwo - endingOne);
+
+    // Orchard x Parent Tree
+    orchardParentTreeDto.setParentTrees(findAllParentTree(orchard.get().getId(), spuId, endingTwo));
+
+    long ending = Instant.now().toEpochMilli();
+    log.info("Time elapsed final: {}", ending - starting);
+    return Optional.of(orchardParentTreeDto);
+  }
+
+  private List<ParentTreeDto> findAllParentTree(String orchardId, Long spuId, long milli) {
+    List<ParentTreeDto> parentTreeDtoList = new ArrayList<>();
+    List<ParentTreeOrchard> parentTreeOrchards =
+        parentTreeOrchardRepository.findAllByOrchardId(orchardId);
+    long endingThree = Instant.now().toEpochMilli();
+    log.info("Time elapsed querying all parent tree to the orchard: {}", endingThree - milli);
+
+    List<Long> orchardIdList =
+        parentTreeOrchards.stream().map(ParentTreeOrchard::getParentTreeId).toList();
+
+    long endingFour = Instant.now().toEpochMilli();
+    log.info("Time elapsed mapping all parent tree orchard ids: {}", endingFour - endingThree);
+
+    List<ParentTree> parentTreeList = parentTreeRepository.findAllIn(orchardIdList);
+
+    long endingFive = Instant.now().toEpochMilli();
+    log.info("Time elapsed finding all parent tree (select in): {}", endingFive - endingFour);
+
+    List<Long> parentTreeIdList = parentTreeList.stream().map(ParentTree::getId).toList();
+    long endingSix = Instant.now().toEpochMilli();
+    log.info("Time elapsed mapping all parent tree ids: {}", endingSix - endingFive);
+
+    List<ParentTreeGeneticQualityDto> qualityDtoList =
+        findAllParentTreeGeneticQualities(spuId, parentTreeIdList);
+    long endingSeven = Instant.now().toEpochMilli();
+    log.info("Time elapsed querying all parent tree genetic quality: {}", endingSeven - endingSix);
+
+    for (ParentTree parentTree : parentTreeList) {
+      ParentTreeDto parentTreeDto = new ParentTreeDto();
+      parentTreeDto.setParentTreeId(parentTree.getId());
+      parentTreeDto.setParentTreeNumber(parentTree.getParentTreeNumber());
+      parentTreeDto.setParentTreeRegStatusCode(parentTree.getParentTreeRegStatusCode());
+      parentTreeDto.setLocalNumber(parentTree.getLocalNumber());
+      parentTreeDto.setActive(parentTree.getActive());
+      parentTreeDto.setTested(parentTree.getTested());
+      parentTreeDto.setBreedingProgram(parentTree.getBreedingProgram());
+      parentTreeDto.setFemaleParentTreeId(parentTree.getFemaleParentParentTreeId());
+      parentTreeDto.setMaleParentTreeId(parentTree.getMaleParentParentTreeId());
+
+      // remove from for
```

Comment:
Is this a `TODO`/`FIXME`? If so, we'd be better off marking it as such:

```suggestion
      // TODO: remove from for
```

Reply:
Oops. Forgot to remove. Done!